File: 02_Arrays_OddOccurrencesInArray.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(a)
    # write your code in Ruby 2.2
    hash = Hash.new(0)
    a.each do |aa|
      hash[aa] += 1
    end 
  
    hash.keys.each do |key|
      if hash[key] % 2 == 1 
        return key 
      end 
    end
  end
  

File: Python/02_Arrays_OddOccurrencesInArray.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    # write your code in Python 3.6
    dic = {}
    for a in A:
        if a in dic: dic[a] += 1
        else: dic[a] = 1
    
    for key in dic.keys():
        if dic[key] % 2 == 1:
            return key 
# end of func
